logstash-core/lib/logstash/pipeline.rb

```diff
@@ -307,18 +306,37 @@ def output_batch(batch)
     outputs_events = batch.reduce(Hash.new { |h, k| h[k] = [] }) do |acc, event|
       # We ask the AST to tell us which outputs to send each event to
       # Then, we stick it in the correct bin
-
-      # output_func should never return anything other than an Array but we have lots of legacy specs
-      # that monkeypatch it and return nil. We can deprecate  "|| []" after fixing these specs
-      outputs_for_event = output_func(event) || []
-
+      outputs_for_event = with_pipeline_rescue(event) { output_func(event) }
       outputs_for_event.each { |output| acc[output] << event }
       acc
     end
 
     # Now that we have our output to event mapping we can just invoke each output
     # once with its list of events
-    outputs_events.each { |output, events| output.multi_receive(events) }
+    outputs_events.each do |output, events|
+      with_pipeline_rescue(events) do
+        output.multi_receive(events)
+      end
+    end
+  end
+
+  def with_pipeline_rescue(event_or_events)
+    yield
+  rescue Exception => e
```

Comment:
Do we really want to rescue `Exception` and not `StandardError`?

See https://robots.thoughtbot.com/rescue-standarderror-not-exception